swagger.json

```diff
@@ -2574,9 +2574,234 @@
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                }
+            }
         }
     },
     "definitions": {
+        "Address": {
+            "type": "object",
+            "description": "Information about an address/geofence. Geofences are either a circle or a polygon.",
+            "properties": {
+                "id": {
+                    "type": "integer",
+                    "description": "ID of the address",
+                    "format": "int64",
+                    "example": 123
+                },
+                "name": {
+                    "type": "string",
+                    "description": "Name of the address or geofence",
+                    "example": "Samsara HQ"
+                },
+                "formattedAddress": {
+                    "type": "string",
+                    "description": "The full address associated with this address/geofence, as it might be recognized by maps.google.com",
+                    "example": "350 Rhode Island St, San Francisco, CA"
+                },
+                "geofence": {
+                    "$ref": "#/definitions/AddressGeofence"
+                }
+            }
+        },
+        "AddressGeofence": {
+            "type": "object",
+            "description": "The geofence that defines this address and its bounds. This can either be a circle, or a polygon - only one key should be provided, depending on the geofence type.",
+            "properties": {
+                "circle": {
+                    "type": "object",
+                    "description": "Information about a circular geofence. This field is only populated if the geofence is a circle.",
+                    "properties": {
+                        "latitude": {
+                            "type": "number",
+                            "format": "float",
+                            "description": "The latitude of the center of the circular geofence",
+                            "example": "37.765363"
+                        },
+                        "longitude": {
+                            "type": "number",
+                            "format": "float",
+                            "description": "The longitude of the center of the circular geofence",
+                            "example": "-122.403098"
+                        }
+                    }
+                },
+                "polygon": {
+                    "type": "object",
+                    "description": "Information about a polygon geofence. This field is only populated if the geofence is a polygon.",
+                    "properties": {
+                        "vertices": {
+                            "type": "array",
+                            "description": "The vertices of the polygon geofence. These geofence vertices describe the perimeter of the polygon, and must consist of at least 3 vertices and less than 40.",
```

Comment:
This should also mention the order that the vertices should be added (e.g. clockwise).

Reply:
Our code iiuc doesn’t care if it is clockwise or not (you can easily add a geofence in a counter-clockwise manner)
Maybe wed want to specify for clarity?

Reply:
I thought the clockwise format was the format that Google requires? It's possible that Google Maps sorts the vertices before rendering the polygon, but I'm not sure.

Reply:
Are the first and last vertices required to be the same?

Reply:
Let me check!

Reply:
I pulled my org's addresses down from the API and it all vertices of the polygon for my drawn address are unique.

Reply:
Oh god, these all submitted after the GitHub outage was over... wow